### DISCUSSION TITLE
Parallel tasks in workflows

### DISCUSSION THREAD

**Discussion by sans712** [codingjoe/joeflow#30] [General]
Does joeflow support parallel tasks in workflows and does this provides graphical representation of workflow?

**Comment by codingjoe** [comment]
Hi @sans712,

Thanks for reaching out. I am always happy if people are curious about joeflow.
To answer your question: YES, the framework does support parallel task execution and blocking and non-blocking joins. Blocking means, you'd wait for all incoming edges to complete or just for one in case of non-blocking.

To run tasks in parallel, all you need are two outgoing edges for a node. That can be done by either adding them in the `edges` definition, of if a task is returning two edges.

If hope my answer was helpful. If you have a good example in mind we could use to improve our documentation, I'd be pleased to do so.

Best,
Joe

**Comment by sans712** [comment]
Thank you so much for your reply.
I am working on workflow automation and want to use joefloe .As I am
complete beginner I needed some help with my project if you can help me
with that could you please share your linkedin profile if interested.
Thanks and Regards
Sanskriti

**Reply by codingjoe** [reply]
I am happy to help you. However, the library builds upon many concepts in Django, so knowing the Django framework well might be really helpful. I send you a connection request on LinkedIn in case you need it. If you require any further assistance, feel free to ask your questions here. They will be helpful for other in the future.
